### PR TITLE
Peakq: Split variable definition from replacement strategy

### PR DESCRIPTION
**Pull request by ryantxu**
This PR, updates the template models so that:
1. the variable definition happens outside each query (like in dashboards)
2. the replacement strategy lives as a sibling to each query -- avoiding the target index
3. Expects []string{} rather than a string -- this will let us support multi-value fields when we are ready (can error now)

Also updated the query structure and swagger definition to expect a URL structure identical to dashboards: &var-{varriable}={value}&...

@kylebrandt -- I commented out the functions that actually do the work!  this is enough to show the model change, and you will be faster at making them work if the models look better :)

### CODE REVIEW

File: pkg/apis/peakq/v0alpha1/types.go [hunk 1]
```diff
@@ -37,43 +37,66 @@ type Target struct {
...
+	// does not support ints as map keys
+	Position *Position `json:"position,omitempty"`
+
+	// How values should be interpolated
+	// See: https://grafana.com/docs/grafana/latest/dashboards/variables/variable-syntax/#advanced-variable-format-options
+	// NOTE: the format parameter is not yet supported!
+	Format string `json:"format,omitempty"`
+
+	// Keep track of the values from previous iterations
+	History []ReplacementHistory `json:"history,omitempty"`
```
Comment (ryantxu) [inline pkg/apis/peakq/v0alpha1/types.go:83]: added mostly as a curiosity -- something we could consider later

File: pkg/apis/peakq/v0alpha1/types.go [hunk 2]
```diff
@@ -37,43 +37,66 @@ type Target struct {
...
+	// Positions is a list of where to perform the interpolation
+	// within targets during render.
+	// The first string is the Idx of the target as a string, since openAPI
+	// does not support ints as map keys
+	Position *Position `json:"position,omitempty"`
+
+	// How values should be interpolated
+	// See: https://grafana.com/docs/grafana/latest/dashboards/variables/variable-syntax/#advanced-variable-format-options
+	// NOTE: the format parameter is not yet supported!
+	Format string `json:"format,omitempty"`
```
Comment (ryantxu) [inline pkg/apis/peakq/v0alpha1/types.go:80]: ^^ for now, we should throw an error if this is configured -- the options mostly support what todo with multi-valued fields

File: pkg/apis/peakq/v0alpha1/types.go [hunk 3]
```diff
@@ -37,43 +37,66 @@ type Target struct {
...
-// during render.
-type Position struct {
-	// IndexIdx is the index of the target in The QueryTemplateSpec Targets property.
-	TargetIdx int `json:"targetIdx"`
-
-	// TargetKey is the location of the property within the the target properties.
+// QueryVariable is the definition of a variable that will be interpolated
+// in targets.
+type VariableReplacement struct {
+	// Path is the location of the property within the the target properties.
```
Comment (kylebrandt) [inline pkg/apis/peakq/v0alpha1/types.go:66]: I think VariableReplacement is where `TargetIdx` property will go, Path is the path within the target.

File: pkg/apis/peakq/v0alpha1/types.go [hunk 4]
```diff
@@ -37,43 +37,66 @@ type Target struct {
 	// TODO 2[uint] seems to panic, maybe implement DeepCopy on data.FrameTypeVersion?
 	// DataTypeVersion data.FrameTypeVersion `json:"dataTypeVersion,omitempty"`
 
+	// Variables that will be replaced in the query
+	Variables map[string][]VariableReplacement `json:"variables"`
```
Comment (kylebrandt) [inline pkg/apis/peakq/v0alpha1/types.go:41]: What is the string supposed to be in the map?

File: pkg/apis/peakq/v0alpha1/types.go (hunk 3 above)
Comment (kylebrandt) [reply pkg/apis/peakq/v0alpha1/types.go:66]: Oh I see you are moving them into the target

**Pull request merged**